tests/test_merge_transactions.py

```diff
@@ -200,16 +200,14 @@ def test_merge_transaction(
 ) -> None:
     listdir_mock.return_value = transactions_file_names
     open_mock.return_value.read.side_effect = transactions_list
-    mock_dummy_datetime = "2024-01-01T10:00:00.00000"
-    datetime_mock.now.return_value.isoformat.return_value = mock_dummy_datetime
 
     runner = CliRunner()
     result = runner.invoke(merge_transactions, [])
 
     assert result.exit_code == 0
     assert (
         write_json_mock.call_args.args[0]
-        == f"data/merged_transactions-{mock_dummy_datetime}.json"
+        == "data/merged_transactions_latest.json"
     )
```

Comment:
Since we now have two calls to the function "write_json" we now need to assert both calls. By removing one of the calls we drop test coverage.

Reply:
To test with "mock_dummy_datetime" created on test_merge_transactions.py which composes the file name, the same variable (or an equal one) must be used on the merge_transactions(). I didn't figured out a low coupling solution for that. Any ideas?

Reply:
What do you mean exactly by the "same variable"?

The intention of the mock is to fake real code dependencies. The mocks in the test file will modify the return value of the library "datetime" in the real code, making real code independent from test constants.